generate_build_config.py

```diff
@@ -51,6 +51,61 @@
 - chroot $CHROOT_ROOT apt-get -y update
 """  # noqa: E501
 
+SETUP_CONTENT = b"""\
+#!/bin/sh -eux
+mv /usr/sbin/grub-probe /usr/sbin/grub-probe.dist
+cat <<"PSUEDO_GRUB_PROBE" > /usr/sbin/grub-probe
+#!/bin/sh
+Usage() {
+   cat <<EOF
+Usage: euca-psuedo-grub-probe
+   this is a wrapper around grub-probe to provide the answers for an ec2 guest
+EOF
+}
+bad_Usage() { Usage 1>&2; fail "$@"; }
+
+short_opts=""
+long_opts="device-map:,target:,device"
+getopt_out=$(getopt --name "${0##*/}" \
+   --options "${short_opts}" --long "${long_opts}" -- "$@") &&
+   eval set -- "${getopt_out}" ||
+   bad_Usage
+
+device_map=""
+target=""
+device=0
+arg=""
+
+while [ $# -ne 0 ]; do
+   cur=${1}; next=${2};
+   case "$cur" in
+      --device-map) device_map=${next}; shift;;
+      --device) device=1;;
+      --target) target=${next}; shift;;
+      --) shift; break;;
+   esac
+   shift;
+done
+arg=${1}
+
+case "${target}:${device}:${arg}" in
+   device:*:/*) echo "/dev/sda1"; exit 0;;
+   fs:*:*) echo "ext2"; exit 0;;
+   partmap:*:*) echo "msdos"; exit 0;;
+   abstraction:*:*) echo ""; exit 0;;
+   drive:*:/dev/sda) echo "(hd0)";;
+   drive:*:/dev/sda*) echo "(hd0,1)";;
```

Comment:
Does this work for uefi images?  Or images where boot isn't the first partition?  This maps any partition to (hd0,1) and I'm not sure if we still have that narrow a use case here.  Same for mapping "device:*:/*" to "/dev/sda1". :-/